python/dgl/data/dgl_dataset.py

```diff
@@ -36,10 +36,12 @@ def __init__(self, name, url=None, raw_dir=None, save_dir=None, force_reload=Fal
         if raw_dir is None:
             self._raw_dir = get_download_dir()
         else:
-            self._raw_dir = raw_dir
+            self._raw_dir = os.path.expanduser(raw_dir)
```

Comment:
This changes behavior of raw_dir and makes a strong hidden assumption the dir is under ~.
But it can under /mnt or /data, etc.

```suggestion
            self._raw_dir = raw_dir
```
Let use take care of the path.